### PR TITLE
Fix Pytorch lightning unit test

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Resolve https://github.com/optuna/optuna/issues/4524

## Description of the changes
See https://github.com/optuna/optuna/issues/4524

### CODE REVIEW

**Comment by c-bata** [comment]
@not522 Could you review this PR if you have time?

File: tests/integration_tests/test_pytorch_lightning.py [hunk 1]
```diff
@@ -1,10 +1,6 @@
-from typing import Any
-from typing import cast
-from typing import Dict
-from typing import List
-from typing import Mapping
-from typing import Sequence
-from typing import Union
+from __future__ import annotations
+
+from collections.abc import Sequence
```
Comment (c-bata) [inline tests/integration_tests/test_pytorch_lightning.py:3]: I guess you unintentionally changed the import statement here.

```suggestion
from typing import Sequence
```

**Review by c-bata — COMMENTED** [review]
Thank you for your pull request! Looks almost good to me. I left one suggestion though.

File: tests/integration_tests/test_pytorch_lightning.py (hunk 1 above)
Comment (Alnusjaponica) [reply tests/integration_tests/test_pytorch_lightning.py:3]: I believe it is recommended by [PEP584](https://peps.python.org/pep-0585/).
If it is better to separate type annotation update from this PR I'll revert the change.

**Review by not522 — APPROVED** [review]
LGTM!

**Review by c-bata — APPROVED** [review]
LGTM!

**Pull request merged**